### DISCUSSION TITLE
How to have multiple JobRequestHandler tied to same JobRequest

### DISCUSSION THREAD

**Discussion by jhchee** [jobrunr/jobrunr#721] [Q&A] [answered]
From the documentation, one JobRequest (DTO) is tied to one JobRequestHandler, is there a way to tie multiple jobs to the same JobRequest (DTO).
```java
public class MyJobRequest implements JobRequest {

  private UUID id;

  public MyJobRequest(UUID id) {
    this.id = id;
  }

  @Override
  public Class<MyJobRequestHandler> getJobRequestHandler() {
      return MyJobRequestHandler.class;
  }

  public UUID getId() {
    return id;
  }

}
```

While the job function can have parameter that doesn't implement JobRequest interface e.g.
```java
public void execute(CustomObject someObjectThatDoesntImplementJobRequest)
```

I would like to know what is the benefit of implementing JobRequest.

**Comment by rdehuyss** [comment, marked as answer]
That's not possible and I also don't see a direct use case. It will make your code probably more difficult to understand.

See the [single responsibility principle](https://en.wikipedia.org/wiki/Single-responsibility_principle).
